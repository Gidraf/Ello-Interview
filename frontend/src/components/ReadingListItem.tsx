import React from 'react';
import { Book } from '../types';
import { Card, CardContent, Typography, Button, CardMedia } from '@mui/material';

interface ReadingListItemProps {
  book: Book;
  removeBookFromReadingList: (title: string) => void;
}

const ReadingListItem: React.FC<ReadingListItemProps> = ({ book, removeBookFromReadingList }) => {
  return (
    <Card style={{ marginBottom: '16px' }}>
    <CardMedia
        sx={{ height: 140 }}
        image={require(`../${book.coverPhotoURL}`)}
        title="green iguana"
      />
      <CardContent>
        <Typography variant="h5">{book.title}</Typography>
        <Typography variant="subtitle1">{book.author}</Typography>
        <Button variant="contained" color="secondary" onClick={() => removeBookFromReadingList(book.title)}>
          Remove
        </Button>
      </CardContent>
    </Card>
  );
};

export default ReadingListItem;
