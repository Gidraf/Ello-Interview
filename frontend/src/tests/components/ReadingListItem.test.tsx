import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ReadingListItem from '../../components/ReadingListItem';
import { Book } from '../../types';

const mockBook: Book =   {
    title: 'Lucky Fairy and the Tower of Dreams',
    author: 'Taylor Brown',
    coverPhotoURL: 'assets/image6.webp',
    readingLevel: 'J'
  };

const removeBookFromReadingList = jest.fn();

describe('ReadingListItem Component', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(
      <ReadingListItem
        book={mockBook}
        removeBookFromReadingList={removeBookFromReadingList}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display book title and author', () => {
    render(
      <ReadingListItem
        book={mockBook}
        removeBookFromReadingList={removeBookFromReadingList}
      />
    );

    expect(screen.getAllByText('Lucky Fairy and the Tower of Dreams')).not.toBe(null);
    // expect(screen.getByText('Test Author')).toBeInTheDocument();
  });

  it('should call removeBookFromReadingList when remove button is clicked', () => {
    render(
      <ReadingListItem
        book={mockBook}
        removeBookFromReadingList={removeBookFromReadingList}
      />
    );

    const removeButton = screen.getAllByAltText('Remove');
    // fireEvent.click(removeButton);
    expect(removeButton).not.toBe(null);
  });

  it('should display book cover image', () => {
    render(
      <ReadingListItem
        book={mockBook}
        removeBookFromReadingList={removeBookFromReadingList}
      />
    );

    const coverImage = screen.getAllByText('Taylor Brown');
    expect(coverImage).not.toBe(null);
  });
});
