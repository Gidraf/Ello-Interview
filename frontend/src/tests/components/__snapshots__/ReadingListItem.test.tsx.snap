// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingListItem Component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    style="margin-bottom: 16px;"
  >
    <div
      class="MuiCardMedia-root css-1999db7-MuiCardMedia-root"
      role="img"
      style="background-image: url(image6.webp);"
      title="green iguana"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        Lucky Fairy and the Tower of Dreams
      </h5>
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Taylor Brown
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorSecondary css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Remove
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
